File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/events/EventAnimationListener.java
package com.abubusoft.xenon.animations.events;

import com.abubusoft.xenon.animations.KeyFrame;

public interface EventAnimationListener<K extends KeyFrame> {

	/**
	 * avvio animazione
	 */
	public void onAnimationStart();
	
	/**
	 * termina animazione
	 * 
	 * @param currentFrame
	 */
	public void onAnimationStop(K currentFrame);
	

	/**
	 * metto in pausa
	 */
	void onAnimationPause(K currentFrame, long enlapsedTime);

	/**
	 * resume dell'animazione
	 */
	void onAnimationResume(K currentFrame, long enlapsedTime);
	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/events/EventAnimationListener.kt
package com.abubusoft.xenon.animations.events

import com.abubusoft.xenon.animations.KeyFrame

interface EventAnimationListener<K : KeyFrame?> {
    /**
     * avvio animazione
     */
    fun onAnimationStart()

    /**
     * termina animazione
     *
     * @param currentFrame
     */
    fun onAnimationStop(currentFrame: K)

    /**
     * metto in pausa
     */
    fun onAnimationPause(currentFrame: K, enlapsedTime: Long)

    /**
     * resume dell'animazione
     */
    fun onAnimationResume(currentFrame: K, enlapsedTime: Long)
}